ladybug/skymodel.py

```diff
@@ -0,0 +1,205 @@
+
+    return dir_norm_rad, dif_horiz_rad
+
+
+"""ZHANG-HUANG SOLAR MODEL"""
+# TODO: Split golbal into direct and diffuse using Perez method.
+
+# extraterrestrial solar constant (W/m2)
+irr0 = 1355
+
+# zhang-huang solar model regression constants
+c0, c1, c2, c3, c4, c5, d_coeff, k_coeff = 0.5598, 0.4982, \
+    -0.6762, 0.02842, -0.00317, 0.014, -17.853, 0.843
+
+
+def zhang_huang_solar_model(alt, cloud_cover, relative_humidity,
+                            dry_bulb_present, dry_bulb_t3_hrs, wind_speed):
+    """Calculate solar flux using the Zhang-Huang model.
+
+    Args:
+        alt = A solar altitudes in degrees.
+        cloud_cover: A float value between 0 and 10 that represents the sky cloud cover
+            in tenths (0 = clear; 10 = completely overcast)
+        relative_humidity: A float value between 0 and 100 that represents
+            the relative humidity in percent.
+        dry_bulb_present: A float value that represents the dry bulb
+            temperature at the time of interest (in degrees C).
+        dry_bulb_t3_hrs: A float value that represents the dry bulb
+            temperature at three hours before the time of interest (in degrees C).
+        wind_speed: A float value that represents the wind speed in m\s.
+
+    Returns:
+        dir_ir: A direct normal radiation value in W/m2.
+        diff_ir: A diffuse horizontall radiation value in W/m2.
+    """
+    # start assuming night time
+    glob_ir = 0
+    dir_ir = 0
+    diff_ir = 0
+
+    if alt > 0:
+        # get sin of the altitude
+        sin_alt = math.sin(math.radians(alt))
+
+        # shortened and converted versions of the input parameters
+        cc, rh, n_temp, n3_temp, w_spd = cloud_cover / 10.0, \
+            relative_humidity, dry_bulb_present, dry_bulb_t3_hrs, wind_speed
+
+        # calculate zhang-huang global radiation
+        glob_ir = ((irr0 * sin_alt *
+                    (c0 + (c1 * cc) + (c2 * cc**2) +
+                     (c3 * (n_temp - n3_temp)) +
+                     (c4 * rh) + (c5 * w_spd))) + d_coeff) / k_coeff
+        if glob_ir < 0:
+            glob_ir = 0
+        else:
+            # calculate direct and diffuse solar
+            k_t = glob_ir / (irr0 * sin_alt)
+            k_tc = 0.4268 + (0.1934 * sin_alt)
+            if k_t >= k_tc:
+                k_ds = k_t - ((1.107 + (0.03569 * sin_alt) +
+                               (1.681 * sin_alt**2)) * (1 - k_t)**2)
+            else:
+                k_ds = (3.996 - (3.862 * sin_alt) +
+                        (1.540 * sin_alt**2)) * k_t**3
+            diff_ir = (irr0 * sin_alt * (k_t - k_ds)) / (1 - k_ds)
+            dir_horiz_ir = (irr0 * sin_alt * k_ds * (1 - k_t)) / (1 - k_ds)
+            dir_ir = dir_horiz_ir / math.sin(math.radians(alt))
+
+    return dir_ir, diff_ir
+
+
+"""ENERGYPLUS HORIZONTAL INFRARED INTESDITY MODEL"""
+# stefan-boltzmann constant
+sigma = 5.6697e-8
+
+
+def energyplus_horizontal_infrared(sky_cover, dry_bulb, dew_point):
```

Comment:
do we need energyplus in the name?
If we do can we move it to the end of the name so if someone looks for horizontal the autocomplete suggests this function?

Reply:
I have removed it and just referenced energyplus in the description.